tfhe/src/integer/bigint/algorithms.rs

```diff
@@ -182,25 +225,109 @@ where
     (quotient, remainder)
 }
 
+pub(crate) fn slow_div_signed<T>(numerator: T, divisor: T) -> (T, T)
+where
+    T: SignedNumeric
+        + ShlAssign<u32>
+        + Shl<u32, Output = T>
+        + Shr<u32, Output = T>
+        + BitOrAssign<T>
+        + SubAssign<T>
+        + BitAnd<T, Output = T>
+        + Ord
+        + Neg<Output = T>
+        + Add<T, Output = T>
+        + Sub<T, Output = T>
+        + CastFrom<T::NumericUnsignedType>,
+    T::NumericUnsignedType: CastFrom<T>
+        + ShlAssign<u32>
+        + Shl<u32, Output = T::NumericUnsignedType>
+        + Shr<u32, Output = T::NumericUnsignedType>
+        + BitOrAssign<T::NumericUnsignedType>
+        + SubAssign<T::NumericUnsignedType>
+        + BitAnd<T::NumericUnsignedType, Output = T::NumericUnsignedType>
+        + Ord
+        + CastFrom<T>,
+{
+    assert!(divisor != T::ZERO);
```

Comment:
is there a special rust assert for division by 0 ? if so we can reproduce it here I guess

Reply:
I don't think there is a special macro to assert for division by zero.

However the message when divising by 0 is attempt to divide by zero`. should we add it ?

Reply:
I would try to match the rust behavior in that case, wdyt ?